modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -450,9 +450,10 @@ const FrontendDataSet = ({
 					value={selectedItemsValue.join(',')}
 				/>
 
-				{items?.length ||
-				overrideEmptyResultView ||
-				inlineAddingSettings ? (
+				{(items?.length ||
+					overrideEmptyResultView ||
+					inlineAddingSettings) &&
+				CurrentViewComponent ? (
```

Comment:
With this change, we are rendering `ClayEmptyState`, showing no results found, until the component is ready. This does not look right. Component should be immediately ready, unless we are loading from a remote URL, and then there `componentLoading`, correctly showing loading indicator until the loading is done.

I'll add this case to our sample module and try to replicate.

Reply:
@marcoscv-work, on fist paint the `CurrentViewComponent` is not ready yet cause the the `view` into the `views` prop has no `component` yet, just the `contentRenderer` (`'table'` in our case).   The call to the `requestComponent()` is into a `useEffect`.

https://github.com/liferay-frontend/liferay-portal/blob/c62f96ddf0a32392d944273c5f11d5c2e49062ac/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js#L341-L356

https://github.com/liferay-frontend/liferay-portal/blob/c62f96ddf0a32392d944273c5f11d5c2e49062ac/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js#L202-L225

https://github.com/liferay-frontend/liferay-portal/blob/c62f96ddf0a32392d944273c5f11d5c2e49062ac/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/index.js#L31-L39

We can set the component to be in a loading state until the request is done. Does it sounds good for you?

Reply:
> We can set the component to be in a loading state until the request is done. Does it sound good for you?

The code in `master`, that you quoted above, is already doing this. `componentLoading` is the state variable that determines the loading state. By the time it is `false` [here](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js#L442), `CurrentViewComponent` should always be set. The check in this PR is unnecessary.

If direct inclusion of `<FrontendDataSet>` is buggy, there is something else going on. As I said, I'm working on extending our sample module with this case.